pkg-manager/core/test/catalogs.ts

```diff
@@ -17,37 +16,40 @@ import { testDefaults } from './utils'
  */
 class CatalogTestsController {
   private readonly workspaceDir: string
+  private projects: Record<ProjectId, ProjectManifest>
```

Comment:
Added this in-memory map to keep to track of the most recent `ProjectManifest` since we no longer re-read packages from `@pnpm/workspace.filter-packages-from-dir`.

The `ProjectManifest` can be out of date after a `updateProjectManifest` call.